### ISSUE TITLE
Return type of optimisers 

### ISSUE THREAD

**Issue by casellimarco** [wplohrmann/optimal-transport-rs#9]
We have already a discrepancy (introduced by myself!). In one case we just return the transport, in the other the cost too. Which option do we prefer?

**Comment by wplohrmann** [comment]
Yeah, I noticed that but decided not to do anything about it for now... I think it would make most sense to also return the cost if we compute it as part of the algorithm. If not it should be a separate function or just omitted if the user can easily calculate it. I'm leaning towards the former, perhaps we could have a trait `Metric` with a method `cost` so you can just do
```
p = sinkhorn(a, b, c)
cost = p.cost(c)
```
